.github/workflows/cdelivery-s3.yml

```diff
@@ -0,0 +1,97 @@
+name: CD reusable workflow for S3
+
+# Control when the action will run.
+on:
+  # Allow the workflow to be reusable
+  workflow_call:
+    inputs:
+      build-folder:
+        required: false
+        default: 'build'
+        type: string
+    secrets:
+      aws-access-key-id:
+        required: true
+      aws-secret-access-key:
+        required: true
+      aws-region:
+        required: true
+      aws-s3-bucket-name:
+        required: true
+      cloudfront-distribution:
+        required: true
+      api-host:
+        required: false
+      port:
+        required: false
+      show-notifications:
+        required: false
+      authentication-host:
+        required: false
+      graasp-compose-host:
+        required: false
+      graasp-perform-host:
+        required: false
+      graasp-analyzer-host:
+        required: false
+      ga-measurement-id:
+        required: false
+      hidden-item-tag-id:
+        required: false
+
+env:
+  BUILD_FOLDER: '${{ inputs.build-folder }}'
+
+# This workflow contains one job called: build-deploy
+jobs:
+  build-deploy:
+    name: Build, sync with S3 and deploy to prod
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Setup node
+        id: setup-node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+          check-latest: true
+
+      - name: Yarn build prod
+        id: build-image
+        env: 
+          REACT_APP_API_HOST: ${{ secrets.api-host }}
+          PORT: ${{ secrets.port }}
+          REACT_APP_SHOW_NOTIFICATIONS: ${{ secrets.show-notifications }}
+          REACT_APP_AUTHENTICATION_HOST: ${{ secrets.authentication-host }}
+          REACT_APP_GRAASP_COMPOSE_HOST: ${{ secrets.graasp-compose-host }}
+          REACT_APP_GRAASP_PERFORM_HOST: ${{ secrets.graasp-preform-host }}
+          REACT_APP_GRAASP_ANALYZER_HOST: ${{ secrets.graasp-analyzer-host }}
+          REACT_APP_GA_MEASUREMENT_ID: ${{ secrets.ga-measurement-id }}
+          REACT_APP_HIDDEN_ITEM_TAG_ID: ${{ secrets.hidden-item-tag-id }}
```

Comment:
So these env are the general env variables and you can define more depending on the project?

Reply:
This question is both for apps and ecs deploy.

Reply:
For the S3 workflows these are the env variables needed to perform the yarn build and are specific of each project and app.
For the ECS these are the environment variables specified in the containers